src/main/flight/rpm_filter.c

```diff
@@ -70,46 +70,48 @@ void rpmFilterInit(const rpmFilterConfig_t *config, const timeUs_t looptimeUs)
 {
     motorIndex = 0;
     harmonicIndex = 0;
-    rpmFilter.numHarmonics = 0; // disable RPM Filtering
+    rpmFilter.activeHarmonicsCount = 0; // disable RPM Filtering
 
     // if bidirectional DShot is not available
     if (!useDshotTelemetry) {
         return;
     }
+    
+    for (int n = 0; n < RPM_FILTER_HARMONICS_MAX; n++) {
+        rpmFilter.weights[n] = constrainf(config->rpm_filter_weights[n] / 100.0f, 0.0f, 1.0f);
+        if (rpmFilter.weights[n] > 0.0f) {
+            rpmFilter.activeHarmonicsCount++;
+        }
+    }
 
     // if RPM Filtering is configured to be off
-    if (!config->rpm_filter_harmonics) {
+    if (!isRpmFilterEnabled()) {
         return;
     }
 
     // if we get to this point, enable and init RPM filtering
-    rpmFilter.numHarmonics = config->rpm_filter_harmonics;
     rpmFilter.minHz = config->rpm_filter_min_hz;
     rpmFilter.maxHz = 0.48f * 1e6f / looptimeUs; // don't go quite to nyquist to avoid oscillations
     rpmFilter.fadeRangeHz = config->rpm_filter_fade_range_hz;
     rpmFilter.q = config->rpm_filter_q / 100.0f;
     rpmFilter.looptimeUs = looptimeUs;
 
-    for (int n = 0; n < RPM_FILTER_HARMONICS_MAX; n++) {
-        rpmFilter.weights[n] = constrainf(config->rpm_filter_weights[n] / 100.0f, 0.0f, 1.0f);
-    }
-
     for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
         for (int motor = 0; motor < getMotorCount(); motor++) {
-            for (int i = 0; i < rpmFilter.numHarmonics; i++) {
-                biquadFilterInit(&rpmFilter.notch[axis][motor][i], rpmFilter.minHz * i, rpmFilter.looptimeUs, rpmFilter.q, FILTER_NOTCH, 0.0f);
+            for (int i = 0; i < RPM_FILTER_HARMONICS_MAX; i++) {
+                biquadFilterInit(&rpmFilter.notch[axis][motor][i], rpmFilter.minHz * (i + 1), rpmFilter.looptimeUs, rpmFilter.q, FILTER_NOTCH, rpmFilter.weights[i]);
```

Comment:
IMO this is incorrect - filter is faded when at rpmFilter.notch. It is IMO cleaner to disable it before motor frequency is known
```suggestion
                biquadFilterInit(&rpmFilter.notch[axis][motor][i], rpmFilter.minHz * (i + 1), rpmFilter.looptimeUs, rpmFilter.q, FILTER_NOTCH, 0);
```

Reply:
I agree that initialising with zero weight is logical. However, I was working under the following assumptions: this part is just initialisation; before my changes, the first harmonic was initialised at 0 Hz (which is absurd), and I was always confused about this part. I made these changes based on the idea that it would make it easier for someone trying to read the code to understand what's going on if the signatures were consistent.
However, if you insist, I'm fine with the initial weight of 0.

Reply:
I agree with ledvinap here. We need to be careful because other code parts potentially assume that the RPM filter is deactivated at init (default).